### PR TITLE
magento/magento2#26089: Customer Sharing Options not respected in REST API.

### LINKED ISSUE #26089: Customer Sharing Options not respected in REST API
### Preconditions (*)
1. PHP 7.3.12
2. Magento 2.4-develop

### Steps to reproduce (*)
1. Create multiple websites and store views. For example website1 with storeview1 and website2 with storeview2;
Screenshot from 2020-06-15 15-33-37

2. Go to Admin->Stores->Configuration->Customers->Customer Configuration->Account Sharing Options and set **Share Customer Accounts** to '**Per Website**';
3. Create a user account in website1;
4. Use REST API to login with user and password to retrieve the bearer token
Screenshot from 2020-06-15 15-37-20

5. Read user details with **GET** `/rest/storeview1/V1/customers/me`
6. Read user details with **GET** `/rest/storeview2/V1/customers/me`

### Expected result (*)
1. /rest/storeview1/V1/customers/me should return user information
2. /rest/storeview2/V1/customers/me should return an error (e.g. unauthorized)

### Actual result (*)
1. /rest/storeview1/V1/customers/me returns user information
2. /rest/storeview2/V1/customers/me returns user information

### PR DESCRIPTION
**Pull request by p-bystritsky**
### Fixed Issues (if relevant)

1. Fixes magento/magento2#26089: Customer Sharing Options not respected in REST API.

### Manual testing scenarios (*)

1. See magento/magento2#26089.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Customer/Model/Plugin/CustomerAuthorization.php [hunk 1]
```diff
@@ -53,9 +72,14 @@ public function aroundIsAllowed(
             && $this->userContext->getUserId()
             && $this->userContext->getUserType() === UserContextInterface::USER_TYPE_CUSTOMER
         ) {
-            return true;
-        } else {
-            return $proceed($resource, $privilege);
+            $customer = $this->customerFactory->create()->load($this->userContext->getUserId());
```
Comment (rogyar) [inline app/code/Magento/Customer/Model/Plugin/CustomerAuthorization.php]: The `customerFactory` will return the customer ORM model. Considering this fact, we are calling the deprecated `load` method in order to retrieve the customer entity. 

I would suggest avoiding this approach and use the corresponding resource model for loading the entity instead

**Review by rogyar — CHANGES_REQUESTED** [review]

**Comment by p-bystritsky** [comment]
@magento run all tests

File: app/code/Magento/Customer/Model/Plugin/CustomerAuthorization.php (hunk 1 above)
Comment (p-bystritsky) [reply app/code/Magento/Customer/Model/Plugin/CustomerAuthorization.php]: @rogyar updated

**Review by rogyar — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @rogyar, thank you for the review.
[ENGCOM-7665](https://jira.corp.magento.com/browse/ENGCOM-7665) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Manual testing scenario:**
1. Create multiple websites and store views. For example website1 with storeview1 and website2 with storeview2;
Screenshot from 2020-06-15 15-33-37

2. Go to Admin->Stores->Configuration->Customers->Customer Configuration->Account Sharing Options and set **Share Customer Accounts** to '**Per Website**';
3. Create a user account in website1;
4. Use REST API to login with user and password to retrieve the bearer token
Screenshot from 2020-06-15 15-37-20

5. Read user details with **GET** `/rest/storeview1/V1/customers/me`

**Actual Result:** :heavy_check_mark: `/rest/storeview1/V1/customers/me` returns user information

screenshot_62

6. Read user details with **GET** `/rest/storeview2/V1/customers/me`

**Before:** :heavy_multiplication_x:  `/rest/storeview2/V1/customers/me` returns user information

screenshot_63

**After:** :heavy_check_mark: `/rest/storeview2/V1/customers/me` returns an error (e.g. unauthorized)

screenshot_61

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all tests

**Pull request merged**